src/query.ts

```diff
@@ -456,26 +438,36 @@ export class Query {
    * @param  {String}   queryName   The query_name parameter value to set
    *
    */
-  public queryName(queryName?: QueryTypes): void {
-    if (queryName) {
-      this.setParameter('query_name', queryName);
+  public queryName(queryName?: string): Query {
```

Comment:
Any non-string value will be _coerced_ to a string in the url querystring… so I think this validation is not necessary.

Also, don't _delete_ `this._queryName` (you're destroying the property in the instance), just set it to undefined.

These kind of checks with no distinction between `null` and `undefined` could be:

```typescript
if (queryName == null || `${queryName}`.trim().length === 0) {
  this._queryName = undefined;
} else if (typeof queryName === 'string') {
  this._queryName = queryName;
} else {
  throw…
}
```